### PR TITLE
feat: add imdb ratings

### PR DESCRIPTION
**Pull request by Primdahl26**
# Description

This adds imdb ratings on all media, furthermore it adds filtering and sorting for imdb ratings

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: backend/cli.py [hunk 1]
```diff
@@ -29,6 +33,51 @@
...
+            {"fields": ["imdb_id", "id"], "limit": 200}
+        )
+
+        for media in all_media.results:
+            for rating in ratings:
+                if rating["imdb_id"] == media.imdb_id:
+                    print(media.id)
+                    client.index("media").update_documents(
+                        [{"id": media.id, "imdb_rating": rating["imdb_rating"]}]
+                    )
```
Comment (AndreasPB) [inline backend/cli.py:78]:
```suggestion
        imdb_media = []

        for media in all_media.results:
            for rating in ratings:
                if rating["imdb_id"] == media.imdb_id:
                    print(media.id)
                    imdb_media.append([{"id": media.id, "imdb_rating": rating["imdb_rating"]}])
                    
        client.index("media").update_documents(imdb_media)
```
This should run a lot faster (only 1 MeiliSearch job)

Comment (AndreasPB) [reply backend/cli.py:78]: Could also be a list comp I guess 🤷

File: internal/types.go [hunk 2]
```diff
@@ -88,6 +92,7 @@ func (tv *TV) toMedia() *Media {
 
 	return &Media{
 		Id:                         tvId,
+		ImdbId:                     getImdbID(tv.ExternalIds),
```
Comment (AndreasPB) [inline internal/types.go:95]:
```suggestion
		ImdbId:                     tv.ExternalIds.ImdbId,
```

File: internal/types.go [hunk 3]
```diff
@@ -146,6 +151,14 @@ type MediaIds struct {
 	Ids []string `json:"ids"`
 }
 
+type ExternalIds struct {
+	ImdbId string `json:"imdb_id"`
+}
+
+func getImdbID(externalIds ExternalIds) string {
+	return externalIds.ImdbId
+}
```
Comment (AndreasPB) [inline internal/types.go:160]: Not needed if the above suggestion works

File: internal/types.go [hunk 4]
```diff
@@ -48,6 +50,7 @@ func (movie *Movie) toMedia() *Media {
 
 	return &Media{
 		Id:                         movieId,
+		ImdbId:                     getImdbID(movie.ExternalIds),
```
Comment (AndreasPB) [inline internal/types.go:53]:
```suggestion
		ImdbId:                     movie.ExternalIds.ImdbId,
```

File: internal/types.go (hunk 2 above)
Comment (Primdahl26) [reply internal/types.go]: Done

File: backend/cli.py (hunk 1 above)
Comment (Primdahl26) [reply backend/cli.py]: Done

File: internal/types.go (hunk 3 above)
Comment (Primdahl26) [reply internal/types.go]: Done

File: internal/types.go (hunk 4 above)
Comment (Primdahl26) [reply internal/types.go]: Done

File: backend/app/models.py [hunk 5]
```diff
@@ -34,6 +35,7 @@ class Hit(BaseModel):
     id: str
     title: str
     poster_path: str
+    imdb_rating: str | None
```
Comment (AndreasPB) [inline backend/app/models.py:38]: Remember to run the TypeScript interface script

File: backend/cli.py [hunk 6]
```diff
@@ -29,6 +34,54 @@
...
+        return
+
+    # First we remove the old files
+    if os.path.isdir(dir):
+        for file in os.listdir(dir):
+            os.remove(os.path.join(dir, file))
+
+    with open(f"{dir}title.ratings.tsv.gz", "wb") as f:
+        f.write(res.raw.read())
+        print("Downloaded imdb file")
```
Comment (AndreasPB) [inline backend/cli.py:57]: Typer echo/our success thing

File: backend/tests/test_sort.py [hunk 7]
```diff
@@ -35,3 +35,21 @@ def test_release_date_descending(self):
         hit_release_dates = [hit["release_date"] for hit in search_results["hits"]]
 
         assert sorted(hit_release_dates, reverse=True) == hit_release_dates
+
```
Comment (AndreasPB) [inline backend/tests/test_sort.py:38]: A `test_filter.py` test for the imdb_rating would be nice as well

Assert `>= 5.9` gets 5.9 and the one above or something like that

File: frontend/src/lib/components/details/top_card.svelte [hunk 8]
```diff
@@ -70,18 +74,34 @@
...
         <h2 class="card-title">{title}</h2>
         {#if imdbId}
-          <a
-            href="https://www.imdb.com/{imdbId.startsWith('tt')
-              ? 'title'
-              : 'name'}/{imdbId}"
-          >
-            <div
-              class="badge badge-secondary mx-2 transform
+          {#if imdbId.startsWith("tt")}
```
Comment (AndreasPB) [inline frontend/src/lib/components/details/top_card.svelte:77]: Are there cases of it not starting with tt?

File: frontend/src/lib/components/media_card.svelte [hunk 9]
```diff
@@ -56,6 +56,11 @@
       </h2>
     </figure>
   {/if}
+  {#if media.imdb_rating}
+    <div class="absolute top-0 left-0 mx-1 -mt-1 opacity-85">
```
Comment (AndreasPB) [inline frontend/src/lib/components/media_card.svelte:60]: Not sure if it does anything
```suggestion
    <div class="absolute top-0 left-0 mx-1 mt-1 opacity-85">
```

File: internal/types.go [hunk 10]
```diff
@@ -125,12 +130,13 @@ type Translations struct {
 	InnerTranslations []struct {
 		Iso311661   string `json:"iso_3166_1"`
 		Iso6391     string `json:"iso_639_1"`
-		Name        string `json:"name"`
+		Title       string `json:"title"`
```
Comment (AndreasPB) [inline internal/types.go:133]: Both are probably there depending on the type, just like below in data

**Review by AndreasPB — APPROVED** [review]
LGTM with a few minor changes and a test 🤓

File: backend/app/models.py (hunk 5 above)
Comment (Primdahl26) [reply backend/app/models.py:38]: I added it manually 🤷🏽

File: frontend/src/lib/components/media_card.svelte (hunk 9 above)
Comment (AndreasPB) [reply frontend/src/lib/components/media_card.svelte:60]: Ahh, okay - Didn't know you could do that

File: frontend/src/lib/components/details/top_card.svelte (hunk 8 above)
Comment (AndreasPB) [reply frontend/src/lib/components/details/top_card.svelte:77]: Would you mind adding a comment? I will 100% forget that :D

Comment (Primdahl26) [reply frontend/src/lib/components/details/top_card.svelte:78]: Done

**Pull request merged**